File: Orlando Walking Tours/Orlando Walking Tours/LocationAnnotation.m
//
//  LocationAnnotation.m
//  Orlando Walking Tours
//
//  Created by Andrew Kozlik on 2/10/15.
//  Copyright (c) 2015 Andrew Kozlik. All rights reserved.
//

#import "LocationAnnotation.h"
#import "HistoricLocation.h"

@implementation LocationAnnotation

-(id)initWithHistoricLocation:(HistoricLocation *)location {
    self = [super init];
    
    if (self) {
        self.title = location.locationTitle;
        self.coordinate = CLLocationCoordinate2DMake(location.latitude.doubleValue, location.longitude.doubleValue);
    }
    
    return self;
}

-(MKAnnotationView *)annotationView {
    
    MKAnnotationView *annotationView = [[MKAnnotationView alloc] initWithAnnotation:self reuseIdentifier:@"LocationAnnotation"];
    
    return annotationView;
}

@end

File: Orlando Walking Tours/Annotations/LocationAnnotation.swift
//
//  LocationAnnotation.swift
//  Orlando Walking Tours
//
//  Created by Keli'i Martin on 5/8/16.
//  Copyright © 2016 Code for Orlando. All rights reserved.
//

import UIKit
import MapKit

class LocationAnnotation: NSObject, MKAnnotation
{
    ////////////////////////////////////////////////////////////
    // MARK: - Properties
    ////////////////////////////////////////////////////////////

    let coordinate: CLLocationCoordinate2D
    var title: String? = nil
    var subtitle: String? = nil
    let churchStreetStation = CLLocationCoordinate2DMake(28.540655, -81.381483)

    ////////////////////////////////////////////////////////////
    // MARK: - MKAnnotation
    ////////////////////////////////////////////////////////////

    init(location: HistoricLocation)
    {
        if let latitude = location.latitude?.doubleValue,
            let longitude = location.longitude?.doubleValue
        {
            self.coordinate = CLLocationCoordinate2DMake(latitude, longitude)
        }
        else
        {
            self.coordinate = churchStreetStation
        }

        if let title = location.locationTitle,
            let subtitle = location.address
        {
            self.title = title
            self.subtitle = subtitle
        }
    }
}
